src/elements/src/types.ts

```diff
@@ -0,0 +1,23 @@
+import { NgModuleRef } from '@angular/core';
+import { WithProperties } from '@angular/elements';
+
+export interface SingleSpaAngularElementsOptions {
+  // This has to be tag name, something like `element: 'my-app'`,
+  // because we would want to access this element through DOM API
+  // and set input properties.
+  element: string;
+  bootstrapFunction(): Promise<NgModuleRef<any>>;
+  // It's possible to pass `@Input()` data to Angular custom element
+  // because custom element is basically a virtual component that has
+  // setters for all `@Input()` properties.
+  // Basically directly:
+  // const element = document.querySelector('app-custom-element');
+  // element.someInputProperty = 10;
+  withProperties?<P>(ngModuleRef: NgModuleRef<any>): WithProperties<P> | Promise<WithProperties<P>>;
```

Comment:
Instead of this, I think that we should just put the single-spa custom props as the properties on the element.

Reply:
That has a more broad use case. The thing is that this module will be bootstrapped only once. The user may want to receive some data from the server before rendering custom web component and then set that data as `@Input()` parameters.